### PR TITLE
Implement logic for examples generation

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Implement logic for examples generation

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

several

### CODE REVIEW

File: amazon_cloud_code_generator/cmd/refresh_examples.py [hunk 1]
```diff
@@ -145,10 +107,10 @@ def inject(target_dir, extracted_examples):
         new_content = ""
         in_examples_block = False
         for l in module_path.read_text().split("\n"):
-            if l == "EXAMPLES = r'''":
+            if l == 'EXAMPLES = r"""':
```
Comment (gravesm) [inline amazon_cloud_code_generator/cmd/refresh_examples.py:110]: This doesn't match what's in `templates/header.j2`, so nothing is being added to the examples block. Either this change should be reverted or the template needs to be changed.

Comment (alinabuzachis) [reply amazon_cloud_code_generator/cmd/refresh_examples.py]: @gravesm Ah, done, thanks! I had some local changes and didn't realise.

File: amazon_cloud_code_generator/cmd/refresh_examples.py [hunk 2]
```diff
@@ -1,127 +1,89 @@
...
 import pathlib
+from typing import Dict, List
 import ruamel.yaml
 import yaml
 
 
-def _task_to_string(task):
+def _task_to_string(task: Dict) -> str:
     a = io.StringIO()
     _yaml = ruamel.yaml.YAML()
```
Comment (gravesm) [inline amazon_cloud_code_generator/cmd/refresh_examples.py:13]:
```suggestion
    _yaml = ruamel.yaml.YAML()
    _yaml.width = 160
```
It's not worth blocking the PR on this, but we may want to consider increasing the default width when dumping the yaml here. If you look at the examples for the eks_cluster module, you can see it has split some of the lines. The generated yaml is still valid, but it does look a little weird. I'm not sure what a good value for the width is.

**Review by gravesm — APPROVED** [review]

File: amazon_cloud_code_generator/cmd/refresh_examples.py (hunk 2 above)
Comment (alinabuzachis) [reply amazon_cloud_code_generator/cmd/refresh_examples.py:13]: @gravesm Updated, thank you for that! It seems 160 also works for eks_fargate_profile. We can update it in the future if needed!

**Pull request merged**